Source/Redemption/Private/HealthComponent.cpp

```diff
@@ -63,18 +64,42 @@ void UHealthComponent::Die()
 }
 
 
-
 void UHealthComponent::GameOver()
 {
 	UUserWidget* GameOverWidget = CreateWidget<UUserWidget>(GetWorld(), GameOverWidgetClass);
 	GameOverWidget->AddToViewport();
 }
 
-void UHealthComponent::TakeDamage(AActor* DamagedActor, float Damage, const class UDamageType* DamageType, class AController* InstigatedBy, AActor* DamageCauser)
+void UHealthComponent::TakeDamage(AActor* DamagedActor, float Damage, const UDamageType* DamageType, class AController* InstigatedBy, AActor* DamageCauser)
 {
-	Health -= Damage;
-	if (Health <= 0)
-	{
-		Die();
-	}
-}
\ No newline at end of file
+    Health -= Damage;
+
+    if (Health <= 0)
+    {
+        Die();
+    }
+    else if (Health == 1 && !CurrentVignetteWidget)
+    {
+        UWorld* World = GetWorld();
+        if (World)
+        {
+            CurrentVignetteWidget = CreateWidget<UUserWidget>(World, VignetteWidgetClass);
+            if (CurrentVignetteWidget)
+            {
+                CurrentVignetteWidget->AddToViewport();
+            }
+            else
+            {
+                UE_LOG(LogTemp, Error, TEXT("Failed to create widget"));
+            }
+        }
+       
+    }
+
+
```

Comment:
why the empty lines here between the else if?

Reply:
Just a mistake or something I deleted while coding and didn't remove the spaces most likely.